src/smt/theory_str_noodler/counter_automaton.h

```diff
@@ -17,12 +17,19 @@
 #include "formula.h"
 #include "util.h"
 
+#define INSERT_LEXICOGRAPHIC_BRANCH_ON_CMP_RESULT(cmp_expr) \
+    {                                       \
+        auto cmp_res = cmp_expr;            \
+        if (cmp_res < 0) return true;       \
+        else if (cmp_res > 0) return false; \
+    }
+
```

Comment:
Why maro and not function?

Reply:
Because the function would need to return three values - `Is greater, is smaller, is equal` which would require the caller to essentially have the same logic (sequence of `if`s, etc.). I added the macro mostly when I was changing around the structure members that were being compared (adding/removing) so having some single-line gadget was nice. I can remove it if you wish.

Reply:
Ah ok, so equality means fall trough. Maybe a different name would be better? Looking at further code it seems that `cmp_res` is not used, so you can just use `cmp_expr`, right?

Reply:
The macro is currently named in a way describing what it does: it inserts some branches (ifs) into the function and it is meant to serve to implement lexicographic comparisons. I have added a comment that describes what the macro does, but I am unable to come up with a name that would be more descriptive. Maybe this?
```
#define INSERT_LEXICOGRAPHIC_BRANCH_ON_CMP_RESLUT_WITH_FALLTROUGH_ON_EQ(...)
```
But that is quite long. What do you think?

Reply:
I just realised `CMP` in the name means comparison, and it has nothing to do with the variable `cmp_res` (which I thought the name comes from) :D

Maybe just `BRANCH_ON_CMP_RESULT`? `INSERT` in the name sounds weird in this context.